File: VB/CustomDateEditTest/SpecialDateBorderStyleConverter.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Windows.Markup
Imports System.Windows.Data
Imports DevExpress.Xpf.Editors.Popups.Calendar
Imports System.Windows

Namespace CustomDateEditTest
    Public Class SpecialDateBorderStyleConverter
        Inherits MarkupExtension
        Implements IMultiValueConverter

        Public Function Convert(ByVal values() As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IMultiValueConverter.Convert
            Dim specialDatesList As List(Of Date) = TryCast(values(0), List(Of Date))
            Dim currentCellButton As CalendarCellButton = DirectCast(values(1), CalendarCellButton)
            Dim dt As Date = CDate(DateEditCalendar.GetDateTime(CType(currentCellButton, DependencyObject)))
dt=dt.Date
            Return specialDatesList.Contains(dt)
        End Function

        Public Function ConvertBack(ByVal value As Object, ByVal targetTypes() As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object() Implements IMultiValueConverter.ConvertBack
            Throw New NotImplementedException()
        End Function

        Public Overrides Function ProvideValue(ByVal serviceProvider As IServiceProvider) As Object
            Return Me
        End Function
    End Class
End Namespace

File: CS/CustomDateEditTest/SpecialDateBorderStyleConverter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows.Markup;
using System.Windows.Data;
using DevExpress.Xpf.Editors.Popups.Calendar;
using System.Windows;

namespace CustomDateEditTest {
    public class SpecialDateBorderStyleConverter : MarkupExtension, IMultiValueConverter {

        public object Convert(object[] values, Type targetType, object parameter, System.Globalization.CultureInfo culture) {
            List<DateTime> specialDatesList = values[0] as List<DateTime>;
            CalendarCellButton currentCellButton = (CalendarCellButton)values[1];
            DateTime dt = (DateTime)DateEditCalendar.GetDateTime((DependencyObject)currentCellButton);
dt=dt.Date;
            return specialDatesList.Contains(dt);
        }

        public object[] ConvertBack(object value, Type[] targetTypes, object parameter, System.Globalization.CultureInfo culture) {
            throw new NotImplementedException();
        }

        public override object ProvideValue(IServiceProvider serviceProvider) {
            return this;
        }
    }
}
